oslo/torch/nn/parallel/data_parallel/zero/sharded_optim/_utils.py

```diff
@@ -18,9 +18,9 @@ def is_model_parallel_parameter(p: torch.Tensor):
     Returns:
         bool: True if the parameter is parallel in either mode, False otherwise.
     """
-    parallel_mode = getattr(p, "oslo_parallel", dict())
-    return (
-        ParallelMode.PIPELINE in parallel_mode or ParallelMode.TENSOR in parallel_mode
+    parallel_mode = getattr(p, "oslo_parallel", {})
+    return ParallelMode.PIPELINE in parallel_mode or any(
+        key.startswith("tensor") for key in parallel_mode
```

Comment:
I think directly using the value of enum class `ParallelMode` to compare value is not a good idea.